sql/ui: cluster setting sql.txn_stats.sample_rate of 1 crashes the cluster

Here is the error message:
```SQL
root@localhost:26257/defaultdb> set cluster setting sql.txn_stats.sample_rate = 1;
SET CLUSTER SETTING

Time: 420ms total (execution 420ms / network 0ms)

andrewwoods@MacBook-Pro-3:~/go/src/github.com/cockroachdb/cockroach$ roachprod run local:4 "./workload run tpcc --warehouses=10 --active-warehouses=10 --duration=15m --split --scatter {pgurl:1-3}"
I210208 15:22:50.313766 1 workload/cli/run.go:356  creating load generator...
Initializing 20 connections...
Initializing 100 workers and preparing statements...
I210208 15:22:51.028827 1 workload/cli/run.go:387  creating load generator... done (took 715.065ms)
_elapsed___errors__ops/sec(inst)___ops/sec(cum)__p50(ms)__p95(ms)__p99(ms)_pMax(ms)
    1.0s        0            0.0            0.0      0.0      0.0      0.0      0.0 delivery
    1.0s        0            0.0            0.0      0.0      0.0      0.0      0.0 newOrder
    1.0s        0            0.0            0.0      0.0      0.0      0.0      0.0 orderStatus
    1.0s        0            0.0            0.0      0.0      0.0      0.0      0.0 payment
    1.0s        0            0.0            0.0      0.0      0.0      0.0      0.0 stockLevel
    2.0s        0            0.0            0.0      0.0      0.0      0.0      0.0 delivery
    2.0s        0            0.0            0.0      0.0      0.0      0.0      0.0 newOrder
    2.0s        0            0.0            0.0      0.0      0.0      0.0      0.0 orderStatus
    2.0s        0            0.0            0.0      0.0      0.0      0.0      0.0 payment
    2.0s        0            0.0            0.0      0.0      0.0      0.0      0.0 stockLevel
Error: error in orderStatus: select order fail: EOF
Error: COMMAND_PROBLEM: exit status 1
(1) COMMAND_PROBLEM
Wraps: (2) Node 4. Command with error:
  | ```
  | ./workload run tpcc --warehouses=10 --active-warehouses=10 --duration=15m --split --scatter {pgurl:1-3}
  | ```
Wraps: (3) exit status 1
Error types: (1) errors.Cmd (2) *hintdetail.withDetail (3) *exec.ExitError
```
Repro steps using roachprod local and tpc-c:
```
Use custom binary Alfonso provided 
roachprod create local
chmod +x  ~/Downloads/cockroach
roachprod put local ~/Downloads/cockroach
roachprod put local ./bin/workload
roachprod start local
roachprod run local:4 -- "./cockroach workload fixtures import tpcc --warehouses=10 --db=tpcc"
roachprod run local:4 "./workload run tpcc --warehouses=10 --active-warehouses=10 --duration=15m --split --scatter {pgurl:1-3}"
```

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0xe8 pc=0x430fd06]

goroutine 66224 [running]:
panic(0x7d456a0, 0xb8f07d0)
	/usr/local/opt/go/libexec/src/runtime/panic.go:1064 +0x545 fp=0xc001ca3810 sp=0xc001ca3748 pc=0x4038c65
runtime.panicmem(...)
	/usr/local/opt/go/libexec/src/runtime/panic.go:212
runtime.sigpanic()
	/usr/local/opt/go/libexec/src/runtime/signal_unix.go:742 +0x3f3 fp=0xc001ca3840 sp=0xc001ca3810 pc=0x404fbd3
github.com/cockroachdb/errors/errbase.getPkgPath(0x0, 0x0, 0x49babc9, 0x8)
	/Users/asubiotto/go/src/github.com/cockroachdb/cockroach/vendor/github.com/cockroachdb/errors/errbase/encode.go:235 +0x26 fp=0xc001ca3870 sp=0xc001ca3840 pc=0x430fd06
github.com/cockroachdb/errors/errbase.getFullTypeName(0x0, 0x0, 0xc004060500, 0xa0)
	/Users/asubiotto/go/src/github.com/cockroachdb/cockroach/vendor/github.com/cockroachdb/errors/errbase/encode.go:223 +0x71 fp=0xc001ca38f8 sp=0xc001ca3870 pc=0x430fc31
github.com/cockroachdb/errors/errbase.getTypeDetails(0x0, 0x0, 0x0, 0x30, 0xde57560, 0xc00503ea80, 0x16b80d50, 0xc000500000, 0x0)
	/Users/asubiotto/go/src/github.com/cockroachdb/cockroach/vendor/github.com/cockroachdb/errors/errbase/encode.go:191 +0xad fp=0xc001ca3958 sp=0xc001ca38f8 pc=0x430f9ed
github.com/cockroachdb/errors/errbase.encodeLeaf(0x91083a0, 0xc00503ea80, 0x0, 0x0, 0x10, 0x7dfd260)
	/Users/asubiotto/go/src/github.com/cockroachdb/cockroach/vendor/github.com/cockroachdb/errors/errbase/encode.go:49 +0x1c5 fp=0xc001ca3a30 sp=0xc001ca3958 pc=0x430ee65
github.com/cockroachdb/errors/errbase.EncodeError(0x91083a0, 0xc00503ea80, 0x0, 0x0, 0x0, 0x0)
	/Users/asubiotto/go/src/github.com/cockroachdb/cockroach/vendor/github.com/cockroachdb/errors/errbase/encode.go:37 +0xc8 fp=0xc001ca3a80 sp=0xc001ca3a30 pc=0x430ec68
github.com/cockroachdb/errors.EncodeError(...)
	/Users/asubiotto/go/src/github.com/cockroachdb/cockroach/vendor/github.com/cockroachdb/errors/errbase_api.go:40
github.com/cockroachdb/cockroach/pkg/sql/execinfrapb.NewError(0x91083a0, 0xc004476de0, 0x0, 0x0, 0x91083a0)
	/Users/asubiotto/go/src/github.com/cockroachdb/cockroach/pkg/sql/execinfrapb/data.go:145 +0xfc fp=0xc001ca3af0 sp=0xc001ca3a80 pc=0x5cd7d5c
github.com/cockroachdb/cockroach/pkg/sql/execinfrapb.LocalMetaToRemoteProducerMeta(0x91083a0, 0xc004476de0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/Users/asubiotto/go/src/github.com/cockroachdb/cockroach/pkg/sql/execinfrapb/data.go:324 +0x169 fp=0xc001ca3b48 sp=0xc001ca3af0 pc=0x5cd8609
github.com/cockroachdb/cockroach/pkg/sql/colflow/colrpc.(*Outbox).sendMetadata(0xc004656990, 0x91083a0, 0xc004476de0, 0x175c2608, 0xc00244aed0, 0x0, 0x0, 0x0, 0x0)
	/Users/asubiotto/go/src/github.com/cockroachdb/cockroach/pkg/sql/colflow/colrpc/outbox.go:287 +0x278 fp=0xc001ca3de0 sp=0xc001ca3b48 pc=0x6d16fd8
github.com/cockroachdb/cockroach/pkg/sql/colflow/colrpc.(*Outbox).runWithStream(0xc004656990, 0x91083a0, 0xc004476de0, 0x175c2608, 0xc00244aed0, 0xc00244ae60)
	/Users/asubiotto/go/src/github.com/cockroachdb/cockroach/pkg/sql/colflow/colrpc/outbox.go:326 +0x170 fp=0xc001ca3e58 sp=0xc001ca3de0 pc=0x6d17550
github.com/cockroachdb/cockroach/pkg/sql/colflow/colrpc.(*Outbox).Run(0xc004656990, 0x91082e0, 0xc00be86940, 0x906b700, 0xc00013ba90, 0xe64c26bf00000002, 0xbe63ed9bb74b083a, 0xc7508a49, 0x1, 0xc00244ae60, ...)
	/Users/asubiotto/go/src/github.com/cockroachdb/cockroach/pkg/sql/colflow/colrpc/outbox.go:177 +0x2f7 fp=0xc001ca3ef8 sp=0xc001ca3e58 pc=0x6d16937
github.com/cockroachdb/cockroach/pkg/sql/colflow.(*vectorizedFlowCreator).setupRemoteOutputStream.func1(0x91082e0, 0xc00be86540, 0xc00244ad10)
	/Users/asubiotto/go/src/github.com/cockroachdb/cockroach/pkg/sql/colflow/vectorized_flow.go:655 +0x125 fp=0xc001ca3f90 sp=0xc001ca3ef8 pc=0x6d20805
github.com/cockroachdb/cockroach/pkg/sql/colflow.(*vectorizedFlowCreatorHelper).accumulateAsyncComponent.func1.1(0xc004476d20, 0x91082e0, 0xc00be86540, 0xc00244ad10, 0xc008f55758)
	/Users/asubiotto/go/src/github.com/cockroachdb/cockroach/pkg/sql/colflow/vectorized_flow.go:1267 +0x44 fp=0xc001ca3fb8 sp=0xc001ca3f90 pc=0x6d23204
runtime.goexit()
	/usr/local/opt/go/libexec/src/runtime/asm_amd64.s:1374 +0x1 fp=0xc001ca3fc0 sp=0xc001ca3fb8 pc=0x4072041
created by github.com/cockroachdb/cockroach/pkg/sql/colflow.(*vectorizedFlowCreatorHelper).accumulateAsyncComponent.func1
	/Users/asubiotto/go/src/github.com/cockroachdb/cockroach/pkg/sql/colflow/vectorized_flow.go:1266 +0x6f
```

--- a/pkg/sql/execinfrapb/data.go
+++ b/pkg/sql/execinfrapb/data.go
@@ -17,6 +17,7 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/roachpb"
 	"github.com/cockroachdb/cockroach/pkg/sql/catalog/colinfo"
 	"github.com/cockroachdb/cockroach/pkg/sql/sem/tree"
+	"github.com/cockroachdb/cockroach/pkg/util"
 	"github.com/cockroachdb/cockroach/pkg/util/encoding"
 	"github.com/cockroachdb/cockroach/pkg/util/tracing/tracingpb"
 	"github.com/cockroachdb/errors"
@@ -308,10 +309,12 @@ func LocalMetaToRemoteProducerMeta(
 		rpm.Value = &RemoteProducerMetadata_Metrics_{
 			Metrics: meta.Metrics,
 		}
-	} else {
+	} else if meta.Err != nil {
 		rpm.Value = &RemoteProducerMetadata_Error{
 			Error: NewError(ctx, meta.Err),
 		}
+	} else if util.CrdbTestBuild {
+		panic("unhandled field in local meta or all fields are nil")
 	}
 	return rpm
 }

